source/codegen/service_helpers.py

```diff
@@ -55,6 +55,33 @@ def create_args_for_callback(parameters):
     ])
 
 
+def create_standard_arg(parameter):
+    parameter_name = common_helpers.camel_to_snake(parameter['cppName'])
+    is_array = common_helpers.is_array(parameter['type'])
+    is_output = common_helpers.is_output_parameter(parameter)
+    if is_output and common_helpers.is_string_arg(parameter):
+        type_without_brackets = common_helpers.get_underlying_type_name(
+            parameter['type'])
+        return f'({type_without_brackets}*){parameter_name}.data(), '
+    elif parameter['type'] in {"ViBoolean[]", "ViSession[]", "ViInt16[]"}:
+        return f'{parameter_name}.data(), '
+    elif 'callback_params' in parameter:
+        return f'CallbackRouter::handle_callback, '
+    elif 'callback_token' in parameter:
+        return f'handler->token(), '
+    elif not is_output and common_helpers.is_pointer_parameter(parameter) and 'hardcoded_value' not in parameter:
+        return f'&{parameter_name}_copy, '
+    elif common_helpers.is_enum(parameter) and parameter["type"] == 'char[]':
```

Comment:
This is a new case. For enums that are strings, we get a temporary variable that's a `std::string`, so we need to pass in `.data()`. See `InputStringValuedEnum()` in `ni_fake_non_ivi_service.cpp` for an example of how this all looks in the generated code.

Reply:
Why does this check `parameter["type"] == 'char[]'` instead of `grpc_type == string`?

Per offline discussion, I think the `const` cast will no longer be necessary when we have everything on C++17 (https://en.cppreference.com/w/cpp/string/basic_string/data). Consider a comment here on that.

Reply:
Yeah, I was able to refactor things a bit here to share this logic so this case is gone. Thanks!